packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -755,12 +755,46 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
     );
   }
 
-  // Returns the delta that should result from applying [event] with axis and
-  // direction taken into account.
+  // Returns the delta that should result from applying [event] with axis,
+  // direction, and any modifier specified by the ScrollBehavior taken into
+  // account for horizontal scrollers.
   double _pointerSignalEventDelta(PointerScrollEvent event) {
-    double delta = widget.axis == Axis.horizontal
-      ? event.scrollDelta.dx
-      : event.scrollDelta.dy;
+    final TargetPlatform platform = _configuration.getPlatform(context);
+    final bool horizontal = widget.axis == Axis.horizontal;
+
+    final Set<LogicalKeyboardKey> pressed = HardwareKeyboard.instance.logicalKeysPressed;
+    final bool keyModifiersAreActive = pressed.any(_configuration.pointerAxisModifiers.contains);
+    pressed.removeAll(_configuration.pointerAxisModifiers);
+    // Ensure no other keys are currently pressed so as to not conflict with
+    // other shortcuts.
+    final bool axisIsFlipped = keyModifiersAreActive & pressed.isEmpty;
+    double delta = horizontal ? event.scrollDelta.dx : event.scrollDelta.dy;
+
+    if (horizontal & axisIsFlipped) {
+      switch(platform) {
+        case TargetPlatform.macOS:
+          // Mac already handles shift to flip the input axis, so if the
```

Comment:
this shouldn't be based on the platform from the scroll behaviour, it should be based on the platform of the embedder. This is because this behaviour shouldn't change when you're emulating another platform. I think you want something like Platform.isMacOS here.

(Consider what happens when you're on Windows, and you shift-mousewheel while pretending to be on macOS so you get the macOS bounce and do son. It should still scroll horizontally, just like it would on macOS.)

Reply:
`defaultTargetPlatform` is also about what OS we're simulating. In this case we need to know literally what embedder we're using. (This is the case the API docs refer to when they say "The dart:io.Platform object should only be used directly when it's critical to actually know the current platform, without any overrides possible (for example, when a system API is about to be called)".)

Not sure how you can tell what platform you're on on web though...